cds_ils/ldap/api.py

```diff
@@ -270,26 +272,43 @@ def delete_users(dry_run=True):
     # get all Invenio remote accounts and prepare a list with needed info
     invenio_users = remap_invenio_users(log_func)
 
-    for invenio_user in invenio_users:
+    for remote_account in invenio_users:
+        try:
+            invenio_user = InvenioUser(remote_account)
+        except NoResultFound:
+            continue
+
         ldap_user = ldap_users_map.get(
-            invenio_user["remote_account_person_id"]
-        )
+                invenio_user.data["remote_account_person_id"]
+            )
+
         if not ldap_user:
             # the user in Invenio does not exist in LDAP, delete it
 
             # fetch user and needed values before deletion
-            user_id = invenio_user.user_id
-            user = User.query.filter_by(id=user_id).one()
+            user_id = remote_account.user_id
+            success = False
+            ready_to_delete = True
+
+            if mark_for_deletion:
+                ready_to_delete = False
+                checks = invenio_user.mark_for_deletion()
+                db.session.commit()
```

Comment:
I have the feeling that the `db.session.commit` should go inside the methods `mark/unmark`, WDYT?

Reply:
that was my first impression too, but I followed in the end the other methods for consistency in the code. I will move it.